core/src/Kokkos_Parallel_Reduce.hpp

```diff
@@ -1359,29 +1359,51 @@ struct ParallelReduceAdaptor {
                                   const PolicyType& policy,
                                   const FunctorType& functor,
                                   ReturnType& return_value) {
-    uint64_t kpID = 0;
+    using PassedReducerType = typename return_value_adapter::reducer_type;
```

Comment:
The `FunctorAnalysis` object passed here is the same as we were using before. In case there is no reducer, we assume that the functor implements `init`, `join` and `final`. Passing `Sum` would change behavior by requiring that the type has a  `reduction_identity` specialization.